### PR TITLE
Document all arrow features in docs.rs (#2633)

### LINKED ISSUE #2633: Document All Arrow Features in docs.rs
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

Docs.rs currently only includes the default feature selection, this means it omits optional functionality such as `prettyprint` and `ffi`.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #2633

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow/Cargo.toml [hunk 1]
```diff
@@ -67,6 +67,9 @@ multiversion = { version = "0.6.1", default-features = false }
 bitflags = { version = "1.2.1", default-features = false }
 zstd = { version = "0.11.1", default-features = false, optional = true }
 
+[package.metadata.docs.rs]
+feature = ["prettyprint", "ipc_compression", "dyn_cmp_dict", "ffi", "pyarrow"]
```
Comment (viirya) [inline arrow/Cargo.toml:71]: Seems should be
```suggestion
features = ["prettyprint", "ipc_compression", "dyn_cmp_dict", "ffi", "pyarrow"]
```
?

Comment (tustvold) [reply arrow/Cargo.toml]: Nice spot :sweat_smile:

File: arrow/Cargo.toml [hunk 2]
```diff
@@ -67,6 +67,9 @@ multiversion = { version = "0.6.1", default-features = false }
 bitflags = { version = "1.2.1", default-features = false }
 zstd = { version = "0.11.1", default-features = false, optional = true }
 
+[package.metadata.docs.rs]
```
Comment (alamb) [inline arrow/Cargo.toml:70]: We did something similar with parquet but used --all-features:
https://github.com/apache/arrow-rs/blob/master/parquet/Cargo.toml#L67-L68

I suspect the reason we can't do so in this case is that the `simd` feature will cause the docs build to fail as it doesn't use nightly?

**Review by alamb — APPROVED** [review]
Nice -- 

See https://github.com/apache/arrow-rs/issues/1695 for details of how I tested this previously

Also, this has more info: https://github.com/apache/arrow-rs/pull/1349

File: arrow/Cargo.toml (hunk 2 above)
Comment (tustvold) [reply arrow/Cargo.toml:70]: I didn't want to risk it :sweat_smile:

**Pull request merged**